### PR TITLE
Support C

### PR DESCRIPTION
**Pull request by chlohal**
This adds support for the C language, using the same `cling` interpreter that we use for C++. Code blocks are wrapped in `extern "C" {}` so everything should work sensibly.

I made a different settings-page and executor, but reused most of the setting keys. You can change the path to Cling from the C++ page or the C page, but it will sync between them. The only new setting key which I created was `cUseMain`, since users may want to use a `main()` function for C, but not C++ (or vice versa)

### CODE REVIEW

File: src/executors/CExecutor.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	override run(codeBlockContent: string, outputter: Outputter, cmd: string, cmdArgs: string, ext: string) {
+		const extension = "cpp";
+		const args = `-std=${this.settings.clingStd} ${this.settings.clingArgs}`;
+		
+		codeBlockContent = `extern "C" {
+			${codeBlockContent}
+		}`
+
+		// Run code with a main block
+		if (this.settings.cppUseMain) {
```
Comment (milan338) [inline src/executors/CExecutor.ts:22]: Should this be `this.settings.cUseMain`?

File: src/executors/CExecutor.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+			this.resolveRun = resolve;
+			this.handleChildOutput(child, outputter, undefined);
+		});
+	}
+
+	/**
+	 * Run parent NonInteractiveCodeExecutor handleChildOutput logic, but replace temporary main function name
+	 * In all outputs from stdout and stderr callbacks, from temp_<id>() to main() to produce understandable output
+	 */
+	override async handleChildOutput(child: ChildProcessWithoutNullStreams, outputter: Outputter, fileName: string) {
```
Comment (milan338) [inline src/executors/CExecutor.ts:45]: Is this the same method as in the C++ executor? Is it possible to potentially have this executor inherit from the C++ executor so we don't need to copy this method?

File: src/executors/CExecutor.ts (hunk 1 above)
Comment (chlohal) [reply src/executors/CExecutor.ts:22]: Yes, thanks for catching that!

**Comment by chlohal** [comment]
Interesting; thank you! `ClingExecutor` does seem like the best solution. 

What do you mean by "[having] the C and C++ settings one after the other?"

**Comment by milan338** [comment]
Sorry, I was a bit vague, I was wondering if they could be placed next to each other in the settings menu.

**Comment by chlohal** [comment]
Ah, I see. I've merged them together & made a `ClingExecutor`; this makes the `CppExecutor` rather small, which is great. I'm unsure whether or not `CppExecutor` should have a 1-line `run()` method or not: it seems to work fine without, but it could be good for flexibility. What do you think?

I also made it so that C++ and C have different `arguments` fields; I'm open to discussion on that, but was thinking some more about this, and I thought it good to proactively allow the user to use different compiler flags for different languages

**Comment by milan338** [comment]
Thanks a ton for the work, I think the changes are great. Just a couple of things, when using cling in C mode, it rejects the `std` argument and errors out, so this could be moved out of ClingExecutor into CppExecutor just like you did with the `-x c` arg with C, and the setting should be removed. Also, the `cInject` setting is missing from the `ExecutorSettings` interface in Settings.ts which breaks code injection. Other than that the changes seem to work.

File: src/executors/ClingExecutor.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import NonInteractiveCodeExecutor from './NonInteractiveCodeExecutor';
+import * as child_process from "child_process";
+import type {ChildProcessWithoutNullStreams} from "child_process";
+import type {Outputter} from "src/Outputter";
+import type {ExecutorSettings} from "src/settings/Settings";
+
+export default class ClingExecutor extends NonInteractiveCodeExecutor {
```
Comment (milan338) [inline src/executors/ClingExecutor.ts:7]: Could this be made abstract, since we shouldn't be instantiating it?

Comment (chlohal) [reply src/executors/ClingExecutor.ts:7]: Done! 👍

**Comment by chlohal** [comment]
Done! Thank you so much for your help, Milan! Installing Cling on Windows is difficult & I only have an out-of-date version, so I really appreciate your feedback on my code :) I'm sorry for my latency recently.

**Comment by twibiral** [comment]
This is great!  Thank you both!

**Comment by chlohal** [comment]
Very fair! My uni semester is going into finals soon, so I've also been less active; good luck w/ yours!!

**Comment by twibiral** [comment]
Thank you, good luck for you too :)

**Pull request merged**